### PR TITLE
Add logs and stats in DeleteScheduler

### PR DESCRIPTION
**Pull request by akankshamahajan15**
Summary: Add logs and stats for files marked as trash and files deleted immediately in DeleteScheduler

Test Plan: make check -j64

Reviewers:

Subscribers:

Tasks: T67391953

Tags:

### CODE REVIEW

File: file/delete_scheduler.cc [hunk 1]
```diff
@@ -39,6 +40,7 @@ DeleteScheduler::DeleteScheduler(Env* env, FileSystem* fs,
   assert(sst_file_manager != nullptr);
   assert(max_trash_db_ratio >= 0);
   MaybeCreateBackgroundThread();
+  ROCKS_LOG_INFO(info_log_, "SstFileManager instance: %p", sst_file_manager);
```
Comment (riversand963) [inline file/delete_scheduler.cc:43]: Sorry for the previous confusion. Maybe move this to `DBImpl::Open()`?

File: file/sst_file_manager_impl.h [hunk 2]
```diff
@@ -135,6 +135,11 @@ class SstFileManagerImpl : public SstFileManager {
   // once in the object's lifetime, and before the destructor
   void Close();
 
+  void SetStatisticsPtr(const std::shared_ptr<Statistics>& stats) override {
```
Comment (riversand963) [inline file/sst_file_manager_impl.h:138]: We should call this function in when we open the db.

File: db/db_sst_test.cc [hunk 3]
```diff
@@ -425,6 +428,10 @@ TEST_F(DBSSTTest, RateLimitedDelete) {
   }
   ASSERT_GT(time_spent_deleting, expected_penlty * 0.9);
   ASSERT_LT(time_spent_deleting, expected_penlty * 1.1);
+  ASSERT_EQ(
+      4, options.statistics.get()->getAndResetTickerCount(FILES_MARKED_TRASH));
+  ASSERT_EQ(0, options.statistics.get()->getAndResetTickerCount(
+                   FILES_DELETED_IMMEDIATELY));
```
Comment (riversand963) [inline db/db_sst_test.cc:434]: Nit: options.statistics-> suffices.

**Comment by siying** [comment]
This seems to cause this TSAN failure:

```
    #16 std::__invoke_result<void (*)(void*), rocksdb::BGThreadMetadata*>::type std::__invoke<void (*)(void*), rocksdb::BGThreadMetadata*>(void (*&&)(void*), rocksdb::BGThreadMetadata*&&) /mnt/gvfs/third-party2/libgcc/6ace84e956873d53638c738b6f65f3f469cca74c/7.x/platform007/5620abc/include/c++/7.3.0/bits/invoke.h:95 (db_test+0x000000b8adbf)
    #17 decltype (__invoke((_S_declval<0ul>)(), (_S_declval<1ul>)())) std::thread::_Invoker<std::tuple<void (*)(void*), rocksdb::BGThreadMetadata*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) /mnt/gvfs/third-party2/libgcc/6ace84e956873d53638c738b6f65f3f469cca74c/7.x/platform007/5620abc/include/c++/7.3.0/thread:234 (db_test+0x000000b8adbf)
    #18 std::thread::_Invoker<std::tuple<void (*)(void*), rocksdb::BGThreadMetadata*> >::operator()() /mnt/gvfs/third-party2/libgcc/6ace84e956873d53638c738b6f65f3f469cca74c/7.x/platform007/5620abc/include/c++/7.3.0/thread:243 (db_test+0x000000b8adbf)
    #19 std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(void*), rocksdb::BGThreadMetadata*> > >::_M_run() /mnt/gvfs/third-party2/libgcc/6ace84e956873d53638c738b6f65f3f469cca74c/7.x/platform007/5620abc/include/c++/7.3.0/thread:186 (db_test+0x000000b8adbf)
    #20 execute_native_thread_routine <null> (libstdc++.so.6+0x0000000c86df)
  Previous write of size 8 at 0x7b5c00026730 by main thread:
    #0 std::__shared_ptr<rocksdb::Statistics, (__gnu_cxx::_Lock_policy)2>::operator=(std::__shared_ptr<rocksdb::Statistics, (__gnu_cxx::_Lock_policy)2> const&) /mnt/gvfs/third-party2/libgcc/6ace84e956873d53638c738b6f65f3f469cca74c/7.x/platform007/5620abc/include/c++/7.3.0/bits/shared_ptr_base.h:1036 (db_test+0x0000007441fc)
    #1 std::shared_ptr<rocksdb::Statistics>::operator=(std::shared_ptr<rocksdb::Statistics> const&) /mnt/gvfs/third-party2/libgcc/6ace84e956873d53638c738b6f65f3f469cca74c/7.x/platform007/5620abc/include/c++/7.3.0/bits/shared_ptr.h:93 (db_test+0x0000007441fc)
    #2 rocksdb::DeleteScheduler::SetStatisticsPtr(std::shared_ptr<rocksdb::Statistics> const&) file/delete_scheduler.h:87 (db_test+0x0000007441fc)
    #3 rocksdb::SstFileManagerImpl::SetStatisticsPtr(std::shared_ptr<rocksdb::Statistics> const&) file/sst_file_manager_impl.h:140 (db_test+0x0000007441fc)
    #4 rocksdb::DBImpl::Open(rocksdb::DBOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<rocksdb::ColumnFamilyDescriptor, std::allocator<rocksdb::ColumnFamilyDescriptor> > const&, std::vector<rocksdb::ColumnFamilyHandle*, std::allocator<rocksdb::ColumnFamilyHandle*> >*, rocksdb::DB**, bool, bool) db/db_impl/db_impl_open.cc:1610 (db_test+0x0000007441fc)
    #5 rocksdb::DB::Open(rocksdb::DBOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<rocksdb::ColumnFamilyDescriptor, std::allocator<rocksdb::ColumnFamilyDescriptor> > const&, std::vector<rocksdb::ColumnFamilyHandle*, std::allocator<rocksdb::ColumnFamilyHandle*> >*, rocksdb::DB**) db/db_impl/db_impl_open.cc:1362 (db_test+0x000000746bc8)
    #6 rocksdb::DB::Open(rocksdb::Options const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rocksdb::DB**) db/db_impl/db_impl_open.cc:1339 (db_test+0x000000746dd0)
    #7 rocksdb::DBTestBase::TryReopen(rocksdb::Options const&) db/db_test_util.cc:674 (db_test+0x0000005bf725)
    #8 rocksdb::DBTestBase::Reopen(rocksdb::Options const&) db/db_test_util.cc:628 (db_test+0x0000005bfb5f)
    #9 rocksdb::DBTestRandomized_Randomized_Test::TestBody() db/db_test.cc:3238 (db_test+0x000000558a70)
    #10 void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:3899 (db_test+0x000000d9caa8)
    #11 void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:3935 (db_test+0x000000d9caa8)
    #12 testing::Test::Run() third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:3973 (db_test+0x000000d90439)
    #13 testing::TestInfo::Run() third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:4149 (db_test+0x000000d905e6)
    #14 testing::TestCase::Run() third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:4267 (db_test+0x000000d907d9)
    #15 testing::internal::UnitTestImpl::RunAllTests() third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:6633 (db_test+0x000000d90c9a)
    #16 bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:3899 (db_test+0x000000d9d1c4)
    #17 bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:3935 (db_test+0x000000d9d1c4)
    #18 testing::UnitTest::Run() third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:6242 (db_test+0x000000d91146)
    #19 RUN_ALL_TESTS() third-party/gtest-1.8.1/fused-src/gtest/gtest.h:22104 (db_test+0x00000049492d)
    #20 main db/db_test.cc:6673 (db_test+0x00000049492d)
```
@akankshamahajan15 can you fix it?

**Comment by akankshamahajan15** [comment]
Yes. I will fix this. Thanks.

**Pull request closed without merging**